tasks/uk/census/scotland.py

```diff
@@ -0,0 +1,56 @@
+# http://www.scotlandscensus.gov.uk/ods-web/data-warehouse.html#bulkdatatab
+
+import csv
+import os
+import re
+
+from luigi import Parameter
+
+from tasks.util import shell, DownloadUnzipTask, TempTableTask
+from tasks.meta import current_session
+
+
+class DownloadScotlandLocal(DownloadUnzipTask):
+
+    URL = 'http://www.scotlandscensus.gov.uk/ods-web/download/getDownloadFile.html?downloadFileIds=Output%20Area%20blk'
+
+    def download(self):
+        shell('wget -O {output}.zip {url}'.format(output=self.output().path, url=self.URL))
+
+
+class ImportScotland(TempTableTask):
+    table = Parameter()
+
+    def requires(self):
+        return DownloadScotlandLocal()
+
+    @staticmethod
+    def id_to_column(colid):
+        return re.sub(r'[:/, \-\.\(\)]', '_', '_'.join(colid.split(';')[0].split(':')[-2:]))
+
+    def run(self):
+        with open(os.path.join(self.input().path, self.table + '.csv')) as csvfile:
+            reader = csv.reader(csvfile)
+            header = reader.next()
+
+            cols = [ImportScotland.id_to_column(c) for c in header[1:]]
+
+            session = current_session()
+            session.execute('CREATE TABLE {output} (GeographyCode TEXT PRIMARY KEY, {cols})'.format(
+                output=self.output().table,
+                cols=', '.join(['{} NUMERIC'.format(c) for c in cols])
+            ))
+
+            rows = []
+            for r in reader:
```

Comment:
Why don't we remove the `S92` lines and then use COPY instead of INSERT? How many lines are we talking about?

Reply:
Yeah, it's in the plans. I initially thought that `execute` will batch, but due to how psycopg is implemented, it doesn't 😞.

We don't even need to remove S92, since it is filtered later, so this will eventually become a simple COPY.